lib/services/access_checker.go

```diff
@@ -298,6 +307,32 @@ func (a *accessChecker) CheckAccess(r AccessCheckable, mfa AccessMFAParams, matc
 	return trace.Wrap(a.RoleSet.checkAccess(r, mfa, matchers...))
 }
 
+// GetKubeResources returns the allowed and denied Kubernetes Resources configured
+// for a user.
+func (a *accessChecker) GetKubeResources(cluster types.KubeCluster) ([]types.KubernetesResource, []types.KubernetesResource) {
+	var (
+		allowedResources []types.KubernetesResource
+		deniedResources  []types.KubernetesResource
+	)
+	if len(a.info.AllowedResourceIDs) > 0 {
+		for _, r := range a.info.AllowedResourceIDs {
+			if r.Kind == types.KindKubePod && r.Name == cluster.GetName() && r.ClusterName == a.localCluster {
+				splitted := strings.SplitN(r.SubResourceName, "/", 2)
```

Comment:
Let's check `len(splitted)` before we assume it's 2. (It should, but better to be safe.)